Objects/dictobject.c

```diff
@@ -414,7 +414,7 @@ dictkeys_set_index(PyDictKeysObject *keys, Py_ssize_t i, Py_ssize_t ix)
 
 /* GROWTH_RATE. Growth rate upon hitting maximum load.
  * Currently set to used*3.
- * This means that dicts double in size when growing without deletions,
+ * This means that dicts three times in size when growing without deletions,
```

Comment:
```suggestion
 * This means that dicts triple in size when growing without deletions,
```

Reply:
Yes, for a dict with `N` slots, at most about `2*N/3` can be used.  That's the maximum value of `ma
_used`.  So tripling `ma_used` then gives a dict with `3*(2*N/3) ` = `2*N` slots, only doubling (not tripling) the total size.

Reply:
If so ，
https://github.com/python/cpython/blob/3.5/Objects/dictobject.c
#define GROWTH_RATE(d) (((d)->ma_used*2)+((d)->ma_keys->dk_size>>1))
https://github.com/python/cpython/blob/3.6/Objects/dictobject.c
#define GROWTH_RATE(d) (((d)->ma_used*2)+((d)->ma_keys->dk_size>>1))
(d)->ma_keys->dk_size>>1 === (d)->ma_used*1 ？